test/environments.js

```diff
@@ -182,10 +174,25 @@ class BrowserTypeEnvironment {
         await new Promise(f => setImmediate(f));
         return result;
       };
-      new BrowserTypeDispatcher(dispatcherConnection.rootScope(), this._browserType);
-      overridenBrowserType = await connection.waitForObjectWithKnownName(this._browserType.name());
+      new PlaywrightDispatcher(dispatcherConnection.rootScope(), this._playwright);
+      this.overriddenPlaywright = await connection.waitForObjectWithKnownName('playwright');
     }
-    state.browserType = overridenBrowserType;
+    state.playwright = this.overriddenPlaywright;
+  }
+
+  async afterAll(state) {
+    delete state.playwright;
+  }
+}
+
+class BrowserTypeEnvironment {
+  constructor(playwrightEnvironment, browserName) {
+    this._playwrightEnvironment = playwrightEnvironment;
+    this._browserName = browserName;
+  }
+
+  async beforeAll(state) {
+    state.browserType = this._playwrightEnvironment.overriddenPlaywright[this._browserName];
```

Comment:
Why not `state.playwright[this._browserName]`? You don't need `this._playwrightEnvironment`.

Reply:
Because it is unclear what is in state.playwright!